### PR TITLE
New design and features

### PR DESCRIPTION
**Pull request by DNLHC**
Hey, sorry it took so long. I have also added a few small features, hope you like it.
Here is the final result:
lepton new design
Most of those icons i used are from open-source icon pack [evil-icons](http://evil-icons.io/)

Tag sections are now flexible, height of each section is distributed equally. You can expand one by clicking on it's title, the rest sections will shrink.
Second click on expanded section will lead to default state of all sections.
lepton new design

Snippet control panel is now fixed to the top, so you don't need to scroll back whenever you need to use control buttons. It's helpful when you're managing a big snippet.
lepton new design

### CODE REVIEW

**Comment by hackjutsu** [comment]
@DNLHC 

Thanks for your effort:) I'll review and play with it this week~

File: app/containers/snippet/index.js [hunk 1]
```diff
@@ -433,29 +438,35 @@ class Snippet extends Component {
...
+              title='Share'
+              href='#'
+              onClick={ this.handleShareClicked.bind(this, activeSnippet.brief.html_url) }>
+              <div dangerouslySetInnerHTML={{__html: shareIcon}} />
+              </a>
+            <a className='snippet-control'
+              title='Revisions'
+              href={ activeSnippet.brief.html_url + '/revisions' }>
+              <div dangerouslySetInnerHTML={{__html: eyeIcon}} />
+              </a>
```
Comment (hackjutsu) [inline app/containers/snippet/index.js]: Indention?

File: app/containers/snippet/index.js [hunk 2]
```diff
@@ -433,29 +438,35 @@ class Snippet extends Component {
...
-                </a>
-              : null
-          }
+          <div className='header-controls'>
+            <a className='snippet-control'
+              title='Edit'
+              href='#'
+              onClick={ this.showGistEditorModal.bind(this) }>
+              <div dangerouslySetInnerHTML={{__html: editIcon}} />
+              </a>
```
Comment (hackjutsu) [inline app/containers/snippet/index.js]: Indention?

File: app/containers/snippet/index.js [hunk 3]
```diff
@@ -433,29 +438,35 @@ class Snippet extends Component {
...
+            <a className='snippet-control'
+              title='Share'
+              href='#'
+              onClick={ this.handleShareClicked.bind(this, activeSnippet.brief.html_url) }>
+              <div dangerouslySetInnerHTML={{__html: shareIcon}} />
+              </a>
```
Comment (hackjutsu) [inline app/containers/snippet/index.js]: Indention?

**Comment by hackjutsu** [comment]
Thanks @DNLHC for this awesome PR 💯 

I have merged it and applied some tweaks https://github.com/hackjutsu/Lepton/commit/ddc0f63e885e2528afd422f6449256f6aa467dd2 for the tags displaying .

**Comment by hackjutsu** [comment]
I created a "UX design" label for UX specific issues.  If you are interested, you are welcome to check out these two:)

- https://github.com/hackjutsu/Lepton/issues/153
- https://github.com/hackjutsu/Lepton/issues/154

**Comment by DNLHC** [comment]
@hackjutsu You are more than welcome, I'm glad I could help!
And yeah, i would certainly love to help on these.

Btw i checked your screenshots and saw this gap

it's strange because i don't have this

**Comment by hackjutsu** [comment]
@DNLHC This is a scroll bar handler that will show up on a mouse hover :)

**Comment by DNLHC** [comment]
@hackjutsu 

Oh, then  `overflow: overlay` on container should do the trick.

**Comment by hackjutsu** [comment]
@DNLHC 

Feel free to create an issue and send a PR to fix it:)

**Pull request merged**